src/components/mainpage.js

```diff
@@ -28,8 +30,46 @@ class MainPage extends Component {
                        mode: PageMode.CONNECT,
                        message: 'Attempting to connect to desktop, please wait...' };
 
+
+                       state = {
```

Comment:
This is not defined, you want to add those properties to the state object defined on line 27